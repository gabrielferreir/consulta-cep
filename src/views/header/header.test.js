import React from 'react';
import ConnectHeader, {Header} from './header';
import thunk from 'redux-thunk';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import {configure} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import configureStore from 'redux-mock-store';


const mockStore = configureStore([thunk]);
const initialState = {};

let store;

beforeEach(() => {
    configure({adapter: new Adapter()});
    store = mockStore(initialState);
});

it('Deve renderizar o cabeçalho', () => {
    const wrapper = shallow(<ConnectHeader store={store}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
});

it("Deve atualizar o valor do CEP", () => {
    const updateCep = jest.fn();

    const app = shallow(<Header updateCep={updateCep}/>);

    const input = app.find('input');
    input.simulate('change', { target: { value: '14402-336' } });

    expect(updateCep.mock.calls.length).toBe(1)
});

it("Deve chamar a função que busca o endereço ao apertar Enter", () => {
    const cepRequest = jest.fn();

    const app = shallow(<Header cepRequest={cepRequest}/>);

    const input = app.find('input');
    input.simulate('keydown', {keyCode: 13});

    expect(cepRequest.mock.calls.length).toBe(1)
});

it("Deve chamar a função que busca o endereço ao clicar no botão", () => {
    const cepRequest = jest.fn();

    const app = shallow(<Header cepRequest={cepRequest}/>);

    const input = app.find('button');
    input.simulate('click');

    expect(cepRequest.mock.calls.length).toBe(1)
});

it("Deve exibir a mensagem de erro", () => {
    const error = 'Tivemos um problema, tente novamente mais tarde';
    const app = shallow(<Header error={error}/>);

    expect(toJson(app)).toMatchSnapshot();
});