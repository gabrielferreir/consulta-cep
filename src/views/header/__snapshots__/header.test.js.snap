// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deve chamar a função que busca o endereço ao clicar no botão 1`] = `
<header
  className="header"
>
  <div
    className="header__wrapper-title"
  >
    <h4
      className="header__title"
    >
      Consultar
    </h4>
  </div>
  <div
    className="header__wrapper-field"
  >
    <label
      className="header__label-field"
    >
      CEP
    </label>
    <input
      className="header__field"
      onKeyDown={[Function]}
      placeholder="02050-010"
      type="text"
    />
    <button
      className="header__search-button"
      onClick={[Function]}
    >
      Buscar
    </button>
  </div>
  <span
    className="header__error"
  >
    Tivemos um problema, tente novamente mais tarde
  </span>
</header>
`;

exports[`Deve exibir a mensagem de erro 1`] = `
<header
  className="header"
>
  <div
    className="header__wrapper-title"
  >
    <h4
      className="header__title"
    >
      Consultar
    </h4>
  </div>
  <div
    className="header__wrapper-field"
  >
    <label
      className="header__label-field"
    >
      CEP
    </label>
    <input
      className="header__field"
      onKeyDown={[Function]}
      placeholder="02050-010"
      type="text"
    />
    <button
      className="header__search-button"
      onClick={[Function]}
    >
      Buscar
    </button>
  </div>
  <span
    className="header__error"
  >
    Tivemos um problema, tente novamente mais tarde
  </span>
</header>
`;

exports[`Deve renderizar o cabeçalho 1`] = `
<ContextProvider
  value={null}
>
  <Header
    cepRequest={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    updateCep={[Function]}
  />
</ContextProvider>
`;
